dev/devicelab/lib/framework/ios.dart

```diff
@@ -63,14 +63,16 @@ Future<void> testWithNewIOSSimulator(
     '--json',
   ], workingDirectory: flutterDirectory.path);
 
-  // Get the preferred runtime build for the selected Xcode version. Preferred
+  // First check for userOverriddenBuild, which may be set in CI by mac_toolchain.
+  // Next, get the preferred runtime build for the selected Xcode version. Preferred
   // means the runtime was either bundled with Xcode, exactly matched your SDK
   // version, or it's indicated a better match for your SDK.
   final Map<String, Object?> decodeResult =
       json.decode(runtimesForSelectedXcode) as Map<String, Object?>;
   final String? iosKey =
       decodeResult.keys.where((String key) => key.contains('iphoneos')).firstOrNull;
   final String? runtimeBuildForSelectedXcode = switch (decodeResult[iosKey]) {
+    {'userOverriddenBuild': final String build} => build,
```

Comment:
Xcode 16.2 is weird in that it originally shipped with iOS 18.2 (22C150), but later the "preferredBuild" became `18.3.1` (22D8075). mac_toolchain is still using iOS 18.2 and sets it as the [override build](https://chromium.googlesource.com/infra/infra/+/refs/heads/main/go/src/infra/cmd/mac_toolchain/install.go#457). This will make Flutter tests using a simulator use the override version if set.

This was causing a flaky error: https://ci.chromium.org/ui/p/flutter/builders/try/Mac_arm64%20module_test_ios/3351/overview